presto-main/src/test/java/com/facebook/presto/spiller/TestSpillCipherPagesSerde.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spiller;
+
+import com.facebook.presto.execution.buffer.PagesSerde;
+import com.facebook.presto.execution.buffer.SerializedPage;
+import com.facebook.presto.execution.buffer.TestingPagesSerdeFactory;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.execution.buffer.PageCodecMarker.ENCRYPTED;
+import static com.facebook.presto.operator.PageAssertions.assertPageEquals;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.expectThrows;
+
+public class TestSpillCipherPagesSerde
+{
+    @Test
+    public void testSpillerEncryptedPagesSerde()
+    {
+        SpillCipher cipher = new AesSpillCipher();
+        PagesSerde serde = new TestingPagesSerdeFactory().createPagesSerdeForSpill(Optional.of(cipher));
+        List<Type> types = ImmutableList.of(VARCHAR);
+        Page empty = new Page(VARCHAR.createBlockBuilder(null, 0).build());
+        assertPageEquals(types, serde.deserialize(serde.serialize(empty)), empty);
+
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(null, 2);
+        VARCHAR.writeString(blockBuilder, "hello");
+        VARCHAR.writeString(blockBuilder, "world");
+        Page nonEmpty = new Page(blockBuilder.build());
+
+        SerializedPage serialized = serde.serialize(nonEmpty);
+        assertPageEquals(types, serde.deserialize(serialized), nonEmpty);
+        assertTrue(ENCRYPTED.isSet(serialized.getPageCodecMarkers()), "page should be encrypted");
+
+        cipher.destroy();
+
+        PrestoException serializeException = expectThrows(PrestoException.class, () -> serde.serialize(nonEmpty));
```

Comment:
consider using helper method and checking error message:

```
        assertFailure(() -> serde.serialize(nonEmpty), "Spill cipher already destroyed");
        assertFailure(() -> serde.deserialize(serialized), "Spill cipher already destroyed");
    }

    private static void assertFailure(Assert.ThrowingRunnable runnable, String expectedErrorMessage)
    {
        PrestoException exception = expectThrows(PrestoException.class, runnable);
        assertEquals(exception.getErrorCode(), GENERIC_INTERNAL_ERROR.toErrorCode());
        assertEquals(exception.getMessage(), expectedErrorMessage);
    }
```

Reply:
Will do, thanks.